### ISSUE TITLE
Validation loss computation in multi-device training

### ISSUE THREAD

**Issue by marcopeix** [Nixtla/neuralforecast#1558]
Currently, validation loss computation is not adapted for multi-device training.

For univariate models, the reported validation loss is only partial (we only use the loss from the first device).

For multivariate models, the same loss is computed many times, which is inefficient.

Better to also distribute the validation step, and correctly combine and weigh the loss.
